python/cuml/test/test_one_hot_encoder.py

```diff
@@ -0,0 +1,289 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pytest
+from cudf import DataFrame
+from cuml.preprocessing import OneHotEncoder
+
+import cupy as cp
+import pandas as pd
+import numpy as np
+
+from sklearn.preprocessing import OneHotEncoder as SkOneHotEncoder
+
+from cuml.test.utils import stress_param
+from pandas.util.testing import assert_frame_equal
+
+
+def _from_df_to_array(df):
+    return list(zip(*[df[feature] for feature in df.columns]))
+
+
+def _from_df_to_cupy(df):
+    """Transform char columns to integer columns, and then create an array"""
+    for col in df.columns:
+        if not np.issubdtype(df[col].dtype, np.number):
+            df[col] = [ord(c) for c in df[col]]
+    return cp.array(_from_df_to_array(df))
+
+
+def _convert_drop(drop):
+    if drop is None or drop == 'first':
+        return drop
+    return [ord(x) if isinstance(x, str) else x for x in drop.values()]
+
+
+def _generate_inputs_from_categories(categories=None,
+                                     n_samples=10,
+                                     seed=5060,
+                                     as_array=False):
+    if categories is None:
+        if as_array:
+            categories = {'strings': list(range(1000, 4000, 3)),
+                          'integers': list(range(1000))}
+        else:
+            categories = {'strings': ['Foo', 'Bar', 'Baz'],
+                          'integers': list(range(1000))}
+
+    rd = np.random.RandomState(seed)
```

Comment:
Is there a reason to generate this on CPU?

Reply:
Yes, I initially generated it on GPU but then I realized that converting from `cudf.DataFrame` to `pandas.DataFrame` was way slower than the other way around.